apps/gitness/src/pages-v2/pull-request/__tests__/pull-request-utils.test.ts

```diff
@@ -0,0 +1,253 @@
+  })
+})
+
+describe('checkIfOutdatedSha', () => {
+  it('should return true if SHAs differ', () => {
+    const result = checkIfOutdatedSha(mockReviewedSHA, mockSourceSHA)
+    expect(result).toBe(true)
+  })
+})
+
+describe('capitalizeFirstLetter', () => {
+  it('should capitalize the first letter of a string', () => {
+    const result = capitalizeFirstLetter('hello')
+    expect(result).toBe('Hello')
+  })
+})
+
+describe('findChangeReqDecisions', () => {
+  it('should find change request decisions from entries', () => {
+    const result = findChangeReqDecisions(mockEntriesForChangeReq, 'approve')
+    expect(result).toEqual([
+      {
+        owner_evaluations: [{ review_decision: 'approve' }]
+      },
+      {
+        owner_evaluations: [{ review_decision: 'approve' }]
+      }
+    ])
+  })
+})
+
+describe('findWaitingDecisions', () => {
+  it('should find waiting decisions from entries', () => {
+    const result = findWaitingDecisions(mockEntriesForWaitingDecisions)
+    expect(result).toEqual([])
+  })
+})
+
+describe('normalizeGitFilePath', () => {
+  it('should normalize file paths with spaces', () => {
+    const result = normalizeGitFilePath('path/with space/file.txt')
+    expect(result).toBe('path/with space/file.txt')
+  })
+})
+
+describe('generateAlphaNumericHash', () => {
+  it('should generate an alphanumeric hash of specified length', () => {
+    const result = generateAlphaNumericHash(5)
+    expect(result).toHaveLength(5)
+  })
+})
+
+describe('getErrorMessage', () => {
+  it('should return an error message from an error object', () => {
+    const error = new Error('Test error')
+    const result = getErrorMessage(error)
+    expect(result).toBe('Test error')
+  })
+})
+
+// Mock data for extractInfoForCodeOwnerContent
+const mockPropsForCodeOwnerContent = {
+  approvedEvaluations: [{ reviewer: { display_name: 'John Doe' } as TypesPrincipalInfo, approved: true }],
+  reqNoChangeReq: false,
+  reqCodeOwnerApproval: true,
+  minApproval: 1,
+  reqCodeOwnerLatestApproval: false,
+  minReqLatestApproval: 0,
+  codeOwnerChangeReqEntries: [],
+  codeOwnerPendingEntries: [],
+  latestCodeOwnerApprovalArr: [],
+  latestApprovalArr: [],
+  codeOwnerApprovalEntries: [
+    {
+      owner_evaluations: [
+        { reviewer: { display_name: 'Jane Smith' } as TypesPrincipalInfo, approved: true } as TypesOwnerEvaluation
+      ]
+    }
+  ],
+  changeReqReviewer: 'John Doe',
+  changeReqEvaluations: []
+}
+
+describe('extractInfoForCodeOwnerContent', () => {
+  it('should extract information for code owner content', () => {
+    const result = extractInfoForCodeOwnerContent(mockPropsForCodeOwnerContent)
+    expect(result).toEqual({
+      title: 'Changes approved',
+      statusMessage: 'Changes were approved by code owners',
+      statusColor: 'text-success',
+      statusIcon: 'success',
+      isNotRequired: false
+    })
+  })
+})
+
+describe('changedFileId', () => {
+  it('should return the changed file ID', () => {
```

Comment:
Might be good to show what happens when there are multiple changed files? Or none?